### PR TITLE
Merge Upstream

### PR DESCRIPTION
**Pull request by toad-dev**
Squid did some wizardry to make monitor rendering even faster, and made moving speaker sounds more fluid.

### CODE REVIEW

File: src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPosition.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            buffer.writeDouble( position.x );
+            buffer.writeDouble( position.y );
+            buffer.writeDouble( position.z );
+
+            buffer.writeBoolean( entity.isPresent() );
+            if( entity.isPresent() ) buffer.writeInt( entity.getAsInt() );
+        }
+
+        @Nonnull
+        public SpeakerPosition reify()
```
Comment (SquidDev) [inline src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPosition.java:79]:
```suggestion
        @Environment( EnvType.CLIENT )
        public SpeakerPosition reify()
```

+imports above. See https://github.com/cc-tweaked/CC-Tweaked/commit/be45b718b32fb80f29e1e9ccba06a515bf3144b1.

Sorry, my fault!

**Review by SquidDev — CHANGES_REQUESTED** [review]

File: src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPosition.java (hunk 1 above)
Comment (Merith-TK) [reply src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPosition.java:79]: should this change be made *before* I merge this? I was hoping there would be discussion on it before I merge this

Comment (toad-dev) [reply src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPosition.java:79]: Yes, wait to merge. I will make the change in a few minutes!

**Review by SquidDev — APPROVED** [review]

**Comment by Merith-TK** [comment]
safe to merge now?

**Comment by toad-dev** [comment]
Yes, it's ready.

**Pull request merged**